mlflow/models/rag_signatures.py

```diff
@@ -81,7 +88,7 @@ class StringResponse:
             type=Array(
                 Object(
                     [
-                        Property("role", DataType.string, required=False),
+                        Property("role", DataType.string),
```

Comment:
@chenmoneygithub Are we sure that making these required is safe from a backwards compatibility perspective? cc @sunishsheth2009. Seems risky.

Reply:
Yea good call! i was consulting Sunish and we checked the code, `required` is the current behavior.  These schema files were added by me for clarity, but at my initial testing I wasn't aware of `databricks.agents.deploy` SDK, which caused an error.

Reply:
cc @prithvikannan who has more context on this and @bbqiu who is working on flexible schemas

Reply:
there is a small corner case here with mismatched mlflow versions. in `agents.deploy()` we compare the model's signature to the RAG signatures from MLflow to see if the model is considered "rag compatible". 

usually the MLflow version that is used when logging the model will be the same version installed when running `agents.deploy()`, but it is possible that the user logged with 2.16.2 and then later tries to deploy when 2.17.1 is installed. imo, this is a really small corner case but i think worth calling out

Reply:
Thanks for the inputs! With that context, I think we should proceed with this change?